### DISCUSSION TITLE
SAE Webfrontend speichert nichts...

### DISCUSSION THREAD

**Discussion by Smart71** [camueller/SmartApplianceEnabler#268] [Installation und Betrieb]
Moin,

ich habe mir das Docker Image (2.0.4) auf meinem Synology NAS mal installiert. Der Container startet auch und die Weboberfläche kann auch unter nas.fritz.box:8080 geöffnet werden. Nur wenn man ein Gerät anlegen möchte oder den Broker konfiguriert wird nichts gespeichert.

Wo könnte denn das Problem liegen?

**Comment by Trunks1982** [comment]
Die configs usw muss alles ausserhalb vom docker gespeichert werden.Du musst extra nen Verzeichniss ausserhalb angeben.

**Comment by chibitz** [comment]
Hi,
ich habe das gleiche Problem Docker Image 1.6.15 auf Qnap über Container Station installiert. Was muss ich da noch wo ändern?

Danke Gruß

**Comment by camueller** [comment]
Da würde ich einen Blick in die [Dokumentation](https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/Docker_DE.md#docker-konfiguration) werfen.

Warum sind User selbst bei kostenfreier Software nicht mal gewillt die Zeit für das Lesen der Dokumentation zu investieren? Wenn ich auch so verfahren wäre, gäbe es keinen SAE :-(

**Reply by Smart71** [reply]
Alles klar, dann bleibt mal unter euch. Ich bin hier dann mal raus…

**Reply by kaputti1o1** [reply]
Nicht alle sind Docker Cracks und lesen sich Seitenweise Abhandlungen über die Erstellung eines Containers durch, wenn Synology und QNAP das mit "einem Klick" erledigen. Deine Anmerkung mag berechtigt sein. Dennoch, einfach keine Fehlermeldung auszugeben, ist schlechte UX. Sieh es als sachliche Kritik und versuche es besser zu machen. Keiner kann dich zwingen, aber manchmal hilft auch etwas Selbstkritik und wenn immer die gleichen (blöden) Fragen kommen, dann hilft das mehr, als solche Aussagen in einem Forum. Ich kann Smart71 also durchaus verstehen.

**Reply by camueller** [reply]
Für die bereitgestellten Docker-Images gibt es nur eine Anleitung für `docker` und `docker-compose`. Wenn die Images auf anderen Platformen eingesetzt werden, gibt es keinen Support von mir. Ich bin weit entfernt davon, ein Docker-Crack zu sein, aber es nervt mich inzwischen sehr, wenn nach Support gerufen wird, weil ohne Verständnis von Containern versucht wird, eine Anwendung virtualisiert zu betreiben. Ich unterstütze Virtualisierung schon durch die Bereitstellung der Images, aber es gibt nur Support für `docker` und `docker-compose`. Zu den diversen Virtualisierungsplatformen gibt es genug Tutorials, man muss sich nur die Mühe machen, sich damit zu befassen. Da sollte sich auch finden lassen, wie Volumes erstellt und eingebunden werden.

**Reply by kaputti1o1** [reply]
Ich fürchte du hast mich falsch verstanden.
1) Niemand erwartet von dir kostenfreien Support. Ich sehe das hier als Support-Plattform der Community (User helfen Usern)
2) Ohne Fehlermeldung im Frontend (und wie ich selbst feststellen musste, auch ohne Fehlermeldung in den Logs), wie soll man den Fehler dann von alleine finden? Erwartest du, dass man sich durch den Code wühlt?

Warum ich so auf dem Thema beharre, weil es mir genauso erging und ich es einfach als schlecht programmiert empfinde, wenn mir eine Applikation keinen Fehler ausspuckt. Und ich scheine da nicht der einzige zu sein, wie der Fall von Smart71 anschaulich zeigt.

Wenn es dir hier zu viel wird auf Fragen von Usern zu antworten, kannst du es sein lassen oder diese Diskussionsplattform schließen oder dich der Kritik annehmen und an deinem Code arbeiten. Wofür ich aber kein Verständnis habe, wenn du hier über support suchende User jammerst.

**Reply by camueller** [reply]
Wenn Du es als schlecht programmiert empfindest hast Du sicher Ahnung davon und ich warte dann auf Deinen Pull-Request mit den Code-Verbesserungen.

Ich jammere nicht über User, die hier Support suchen für unterstützte Umgebungen. Ich beklage mich über User, die sich selbst eine Umgebung ausgesucht haben, die sie nicht beherrschen und dann hier unter dem Deckmantel SAE Support suchen. Abgesehen davon bin ich nicht auf Dein Verständnis angewiesen.

**Reply by kaputti1o1** [reply]
Dann bist du auch sicher nicht auf meinen Pull-Request angewiesen. Kritikfähigkeit ist leider keine deiner Stärken. Viel Spaß noch mit dem Projekt.

**Comment by alexb2506** [comment]
Hallo,
habe heute unter Docker auf Synology das aktuelle Paket installiert. Habe einen Ordner zum beschreiben erstellt. Jedoch lassen sich die Eingaben nicht speichern.
Habe mir die Doku durchgeschaut, stehe aber auf dem Schlauch.

ggf. Ist einer bereit mit mir den Fehler zu beseitigen.

ich habe vollen Zugriff auf das Dashboard.

Ziel ist es Shelly‘s zum Datenerfassen zu bewegen und das zum Home Mager 2.0 zu schreiben. Damit dieser die Verbräuche auflistet.

Freue mich über eueren Support.
VG

**Reply by electric-m** [reply]
Hast du /opt/sae/data aus dem Container gemountet?

**Comment by electric-m** [comment]
Auch auf die Gefahr hin, dass ich hier gleich auch Ärger bekomme:

Die Doku habe ich längst gelesen und auch verstanden.
SAE lief auch mit 1.x ohne Problem unter Docker.
Nach Update auf 2.10 kommt keine Verbindung zum MQTT Broker zustande.

Die Einstellungen dazu werden zwar gespeichert und können auch über externe Tools wie MQTT-Explorer validiert werden, die Fehlermeldung auf der Statusseite kommt trotzdem, dass der Broker nicht erreichbar sei.

Freu mich über jeden Tipp.

**Reply by camueller** [reply]
Dieser Fehler wird in der 2.1.0 nur angezeigt, nachdem der MQTT-Broker konfiguriert wurde bis zum nächsten Restart des SAE.
Also einfach den SAE einmal restarten.
Ich habe den Fehler inzwischen für die nächste SAE-Version geixt.

**Reply by electric-m** [reply]
EDIT: Würde gerne den XML Code schön formatieren, kriege es aber nicht hin.

Danke, funktioniert jetzt.
Jezt stimmt auch die XML unter sae-ip:8080/semp wieder.

Verbrauchsdaten sind auch in NodeRED im Dashboard sichtbar, im Sunnyportal kommt allerdings noch nichts an.
Meine mal hier gelsen zu haben, dass es mitunter sehr lange dauert, bis das dort wieder passt?
HomeManager schon neu gestartet.
Im Anlagenlogbuch steht allerdings noch nichts vom SAE.

Docker Log sagt:
` :: Spring Boot ::        (v2.2.2.RELEASE)
09:44:08.639 [main] INFO  o.s.b.w.e.tomcat.TomcatWebServer - Tomcat initialized with port(s): 8080 (http)
09:44:08.653 [main] INFO  o.a.coyote.http11.Http11NioProtocol - Initializing ProtocolHandler ["http-nio-8080"]
09:44:08.654 [main] INFO  o.a.catalina.core.StandardService - Starting service [Tomcat]
09:44:08.655 [main] INFO  o.a.catalina.core.StandardEngine - Starting Servlet engine: [Apache Tomcat/9.0.29]
09:44:09.959 [main] INFO  o.a.c.c.C.[Tomcat].[localhost].[/] - Initializing Spring embedded WebApplicationContext
09:44:09.959 [main] INFO  o.s.web.context.ContextLoader - Root WebApplicationContext: initialization completed in 3165 ms
09:44:10.598 [main] INFO  o.s.s.c.ThreadPoolTaskScheduler - Initializing ExecutorService 'taskScheduler'
09:44:10.624 [main] INFO  o.a.coyote.http11.Http11NioProtocol - Starting ProtocolHandler ["http-nio-8080"]
09:44:10.663 [main] INFO  o.s.b.w.e.tomcat.TomcatWebServer - Tomcat started on port(s): 8080 (http) with context path ''
09:44:11.763 [http-nio-8080-exec-1] INFO  o.a.c.c.C.[Tomcat].[localhost].[/] - Initializing Spring DispatcherServlet 'dispatcherServlet'
09:44:11.763 [http-nio-8080-exec-1] INFO  o.s.web.servlet.DispatcherServlet - Initializing Servlet 'dispatcherServlet'
09:44:11.792 [http-nio-8080-exec-1] INFO  o.s.web.servlet.DispatcherServlet - Completed initialization in 28 ms`

XML sagt:

`<Device2EM xmlns="http://www.sma.de/communication/schema/SEMP/v1">
<DeviceInfo>
<Identification>
<DeviceId>F-11121985-000000000001-00</DeviceId>
<DeviceName>Vitocal 333-G-NC</DeviceName>
<DeviceType>HeatPump</DeviceType>
<DeviceSerial>7537968501007103</DeviceSerial>
<DeviceVendor>Viessmann</DeviceVendor>
</Identification>
<Characteristics>
<MaxPowerConsumption>2000</MaxPowerConsumption>
</Characteristics>
<Capabilities>
<CurrentPower>
<Method>Measurement</Method>
</CurrentPower>
<Timestamps>
<AbsoluteTimestamps>false</AbsoluteTimestamps>
</Timestamps>
<Interruptions>
<InterruptionsAllowed>false</InterruptionsAllowed>
</Interruptions>
<Requests>
<OptionalEnergy>false</OptionalEnergy>
</Requests>
</Capabilities>
</DeviceInfo>
<DeviceStatus>
<DeviceId>F-11121985-000000000001-00</DeviceId>
<EMSignalsAccepted>false</EMSignalsAccepted>
<Status>On</Status>
<PowerConsumption>
<PowerInfo>
<AveragePower>123</AveragePower>
<Timestamp>0</Timestamp>
<AveragingInterval>60</AveragingInterval>
</PowerInfo>
</PowerConsumption>
</DeviceStatus>
</Device2EM>`

Viele Grüße

**Comment by electric-m** [comment]
Ich weiß nicht warum, aber die SEMP URL war nicht mehr korrekt. Sie verwies zwar auf das richtige Gerät, jedoch ein anderes Netzwerkinterface.
Dies habe ich natürlich korrigiert, und SAE wird so auch wieder vom HomeManager gefunden.
Mein einziges Gerät, das nur Verbrauchsdaten übermittelt, zeigt jedoch weiterhin keine Werte im SunnyPortal.

doc/QA_DE.md habe ich durch und scheint soweit alles korrekt. Daher bin ich weiterhin ratlos.
Testweise habe ich mal die ID um eine Stelle verändert. Das "neue" Gerät kann ich dann auch einrichten, es werden jedoch auch hier keine Wert übermittelt.
HomeManager schon mehrfach neugestartet.

**Reply by camueller** [reply]
> Mein einziges Gerät, das nur Verbrauchsdaten übermittelt, zeigt jedoch weiterhin keine Werte im SunnyPortal.

Du hast geprüft, dass die Leistungsaufnahme für dieses Gerät unter /semp korrekt ist?

**Reply by electric-m** [reply]
Ja, ist korrekt.
Nach 15 minütigem Dauer-Aus des HomeManager findet dieser nun das Gerät mit der neuen ID.
Mit der alten ID bekomme ich es leide rnicht zum Laufen. Auch konnte ich das neue Gerät nicht als Austauschgerät angeben, es war nur das Neuanlegen verfügbar. So habe ich halt die historischen Daten nur für das alte Gerät...
